### DISCUSSION TITLE
One sided relations array how to use it with prisma (MongoDB)

### DISCUSSION THREAD

**Discussion by luluhoc** [prisma/prisma#17437] [Q&A]
### Bug description

Hello to all, I have a database that is used with mongoose so the relations are one-sided relations, and I want to use this DB with Prisma in my other project.

Schema
```
model sales {
  id                       String       @id @map("_id")
  createdAt                DateTime     @db.Date
  lignesId                 String[] @map("lignes")
  lignes                   salelignes[] @relation(fields: [lignesId], references: [id])
  updatedAt                DateTime     @db.Date

  @@index([DateVente], map: "DateVente_1")
  @@index([IdMagasin], map: "IdMagasin_1")
  @@index([Statut], map: "Statut_1")
}
```

When I'm trying to create the one-side relation on lignesId array with id's I'm getting 

```
Error parsing attribute "@relation": The relation field `lignes` on Model `sales` must not specify the `fields` or `references` argument in the @relation attribute. You must only specify it on the opposite field `sale` on model `salelignes`.
```

### How to reproduce

DONE

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```
4.9.0
```

**Comment by nurul3101** [comment]
Hello @luluhoc 👋 

What is the cardinality between the sales and salelignes model? Can you please elaborate on what you are trying to achieve?

**Reply by luluhoc** [reply]
> Hello @luluhoc 👋
> 
> What is the cardinality between the sales and salelignes model? Can you please elaborate on what you are trying to achieve?

I'm using mongoose in my other project and instead of having sales id in saleslignes I have lignesId array in sales model with relation to the saleslignes, I want to populate it with prisma as I do with mongoose.

**Reply by nurul3101** [reply]
Can you also share your saleslignes model? And one sale can have multiple salelignes, correct?

**Reply by luluhoc** [reply]
> Can you also share your saleslignes model? And one sale can have multiple salelignes, correct?

Yes it is one(sale) to many(salelignes).

```
model salelignes {
  id                                  String   @id @map("_id")
  Caracteristique                     String?
  CodeArt                             String?
  CodeArtReprise                      Int?
  TypeLigne                           String
  v                                   Int      @map("__v")
  createdAt                           DateTime @db.Date
  updatedAt                           DateTime @db.Date

  @@index([TypeLigne], map: "TypeLigne_1")
}
```

```
model sales {
  id                       String       @id @map("_id")
  createdAt                DateTime     @db.Date
  lignesId                 String[] @map("lignes") // here I have an array of id with relations to salelignes
  lignes                   salelignes[] @relation(fields: [lignesId], references: [id]) // 
  updatedAt                DateTime     @db.Date

  @@index([DateVente], map: "DateVente_1")
  @@index([IdMagasin], map: "IdMagasin_1")
  @@index([Statut], map: "Statut_1")
}
```

**Reply by nurul3101** [reply]
One-way relations are not supported in Prisma at the moment. Please check this feature request:
- https://github.com/prisma/prisma/issues/12725
So salelignes model would need to have a reference to the sales model. One alternative is to create a joining collection (as described in the feature request) that stores the relation between sales and salelignes.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
